### ISSUE TITLE
Use import.meta.vitest instead of globals

### ISSUE THREAD

**Issue by redbmk** [veritem/eslint-plugin-vitest#78]
It looks like the rule adds globals for `it`, `expect`, etc, which are turned off in `vitest` by default and should instead be taken from `import.meta.vitest`.

When I use:

```typescript
if (import.meta.vitest) {
  const { test } = import.meta.vitest;

  test('hello', () => {});
}
```

I don't get any errors from `expect-expect`. If I however do:

```typescript
if (import.meta.vitest) {
  test('hello', () => {});
}
```
I do get the `expect-expect` error, but the tests don't work without configuring `vitest` to use globals (which I'd prefer not to do).

**Comment by redbmk** [comment]
This would be specific to in-source testing. It looks like `import { expect } from 'vitest'` does behave properly, though it would still be nice if the globals weren't automatically set IMO, and if `import.meta.vitest` also worked.

**Comment by veritem** [comment]
So you basically want me to enable linting rules to work inside
```js
if (import.meta.vitest) {
  test('hello', () => {});
}
```

blocks?

**Comment by redbmk** [comment]
@veritem Sorry for the delay - just saw this now.

Well, not exactly - I think that snippet you shared actually does work already. The linting that I'd like added is if you use `const { test } = import.meta.vitest`, like this:

```typescript
if (import.meta.vitest) {
  const { test } = import.meta.vitest;

  test('hello', () => {});
}
```

Currently, that's valid code for `vitest`, but the linting does not work in that case.

See [In-source testing](https://vitest.dev/guide/in-source.html) for the documentation on that.

**Comment by veritem** [comment]
this have been resolved
